app/src/main/java/com/example/taskyy/ui/viewmodels/ReminderViewModel.kt

```diff
@@ -77,34 +99,106 @@ class ReminderViewModel @Inject constructor(
         }
     }
 
-    private fun formatTimeSelected(time: String) {
-        var timeString = time
-        if (timeString == "0:0") {
-            timeString = ("00:00")
-        }
-        if (timeString == "12:0") {
-            timeString = ("12:00")
+    @OptIn(ExperimentalMaterial3Api::class)
+    private fun formatTimeSelected(time: TimePickerState) {
+
+        val localTime = org.joda.time.LocalTime(time.hour, time.minute)
+        val twelveHourDateFormat =
+            DateTimeFormat.forPattern("hh:mm aa") // 12-hour format with AM/PM
+        val formattedTime = twelveHourDateFormat.print(localTime)
+
+        _state.update { it.copy(selectedTime = formattedTime) }
+    }
+
+    private fun save(title: String, description: String) {
+        val newReminder = Reminder(
+            alarmType = _state.value.formattedReminderType,
+            title = title,
+            description = description,
+            timeInMillis = _state.value.formattedReminderTime,
+            id = Random.nextInt(1..10000),
+
+            )
+        viewModelScope.launch {
+            when (val save = agendaRepository.saveReminder(newReminder)) {
+                is Result.Success -> {
+                    eventChannel.send(ReminderEvent.SaveSuccessful)
+                }
+
+                is Result.Error -> {
+                    eventChannel.send(ReminderEvent.SaveFailed(save.error.asUiText()))
+                }
+            }
         }
-        val twelveHourDateFormat = DateTimeFormatter.ofPattern("HH:mm", Locale.ENGLISH)
-        val time24 = LocalTime.parse(timeString, twelveHourDateFormat)
-        val formatter12 = DateTimeFormatter.ofPattern("hh:mm a")
-        val time12 = time24.format(formatter12)
+    }
+
+    private fun formatTimeForDB(selectedTime: String, selectedDate: String) {
+
+        val dateFormat = SimpleDateFormat("dd MMMM yyyy", Locale.getDefault())
+        dateFormat.timeZone = TimeZone.getTimeZone("UTC")
+        val date = dateFormat.parse(selectedDate) ?: Date(0)
+        val dateMillis = date.time
 
-        _state.update { it.copy(selectedTime = time12) }
+        val timeFormat = SimpleDateFormat("HH:mm a", Locale.getDefault())
+        timeFormat.timeZone = TimeZone.getTimeZone("UTC")
+        val time = timeFormat.parse(selectedTime) ?: Date(0)
+        val timeMillis = time.time
+
+        _state.update { it.copy(formattedReminderTime = dateMillis + timeMillis) }
     }
 
-    private fun save() {
-        TODO("Not yet implemented")
+    private fun formatReminderType(reminderType: String, timeToAdd: Int) {
+        val timeToAddInMilis: Long = when (reminderType) {
+            "10 minutes before" -> timeToAdd * 60 * 1000L
+            "30 minutes before" -> timeToAdd * 60 * 1000L
+            "1 hour before" -> timeToAdd * 60 * 60 * 1000L
+            "6 hours before" -> timeToAdd * 60 * 60 * 1000L
+            "1 day before" -> timeToAdd * 24 * 60 * 60 * 1000L
+            else -> {
```

Comment:
For values like `timeToAdd` I'd always also specify the unit you expect, that's otherwise not clear without trying to understand the code. Here you're talking about minutes I think? So it could make sense calling it `minutesToAdd`

Reply:
Well it's whatever value the user has selected. If the user selected 10 minutes before then timeToAdd is minutes but if they selected 1 hour before then the timeToAdd value is hours. Originally I had an Enum value passed in to determine if it was hours or minutes but I re worked it to this because the code seemed simpler to me

Reply:
Hm, that doesn't feel very consistent to me and wasn't clear at all. Why to you need this time to add at all? You could also perfectly work with the Kotlin Duration class:
```
"10 minutes before" -> 10.minutes.toLong(DurationUnit.MILLISECONDS)
...
"6 hours before" -> 6.hours.toLong(DurationUnit.MILLISECONDS)
```